contrib/vagrant/provision-minimal.sh

```diff
@@ -1,20 +1,64 @@
 #!/bin/bash
-set -euo pipefail
-IFS=$'\n\t'
 
-sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
+function setup {
+	set -euo pipefail
+	IFS=$'\n\t'
 
-# patch incompatible with fail-over DNS setup
-SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
-if [[ -f "${SCRIPT}" ]]; then
-    echo "Removing ${SCRIPT}..."
-    rm "${SCRIPT}"
-    sed -i -e '/^options.*$/d' /etc/resolv.conf
-fi
-unset SCRIPT
+	sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
 
-if [ -f /usr/bin/generate_openshift_service ]
-then
-  sudo /usr/bin/generate_openshift_service
-fi
+	# patch incompatible with fail-over DNS setup
+	SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
+	if [[ -f "${SCRIPT}" ]]; then
+	    echo "Removing ${SCRIPT}..."
+	    rm "${SCRIPT}"
+	    sed -i -e '/^options.*$/d' /etc/resolv.conf
+	fi
+	unset SCRIPT
 
+	if [ -f /usr/bin/generate_openshift_service ]
+	then
+	  sudo /usr/bin/generate_openshift_service
+	fi
+}
+
+function installOpenShift {
+	cd /data/src/github.com/openshift/origin
+	echo "-- Starting build, memory:"
+	free -mh
+	# make clean build
+	echo "Starting openshift"
+	sudo `which openshift` start --public-master=localhost &> openshift.log &
+	echo "-- Now starting as new user..."
+	#oc logout
+	#yes "j" | oc login
+	echo "-- now checking who i am and creating project!"
+	echo "sleeping to avoid openshift.local... kubeconfig missing error..."
+	sleep 5
+	sudo -u vagrant whoami
+	sudo chmod +r openshift.local.config/master/openshift-registry.kubeconfig
+	sudo chmod +r openshift.local.config/master/admin.kubeconfig
+	echo "Creating registry.  Sleeping a while first..."
+	sleep 1
+	oadm registry --create --credentials=openshift.local.config/master/openshift-registry.kubeconfig --config=openshift.local.config/master/admin.kubeconfig
+	echo "now creating project"
+	oc="/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc"
+
+	# warning this project needs to be manually deleted after your first 'vagrant up', or it will persist even after destroying vms.
+	# purge the openshift.local.etcd created on your host.
+	sudo -u vagrant $oc login https://localhost:8443 -u=admin -p=admin --config=/data/src/github.com/openshift/origin/openshift.local.config/master/openshift-registry.kubeconfig
+	#sudo -u vagrant $oc login localhost:8443 -u=admin -p=admin --config=/data/src/github.com/openshift/origin/openshift.local.config/master/openshift-registry.kubeconfig
+	sudo -u vagrant $oc new-project project1 --config=/data/src/github.com/openshift/origin/openshift.local.config/master/openshift-registry.kubeconfig || true
```

Comment:
this may or may not fail , depending on if your etcd directory is clean.  if dirty, it will fail.  hence `|| true` after, which avoids this command exiting in failure.